src/main/java/ro/redteam/taskmanagementsystem/services/UserServiceImpl.java

```diff
@@ -37,4 +40,51 @@ public UserDTO createUser(UserDTO userDTO) {
             throw new EmailExistsException("Invalid email");
         }
     }
+
+    @Override
+    public UserDTO getUserById(Long id) {
+        if(userRepository.existsById(id)){
+            return  objectMapper.convertValue(userRepository.findById(id), UserDTO.class);
+        }else{
+            throw new UserNotFoundException("User does not exist!");
+        }
```

Comment:
please use Optional, and only one request to the data base.
if the optional is present than return the mapped object of the user user entity else throw the exception